### PR TITLE
Source Mage GNU/Linux package manager support

### PR DESCRIPTION
**Pull request by vaygr**
I'm glad to present the 'sorcery' module for Ansible.

[Sorcery](http://sourcemage.org/Sorcery) is a package management system for some super-flexible source-based Linux distro called [Source Mage GNU/Linux](http://sourcemage.org). It is one of the [oldest](http://sourcemage.org/linux-timeline.png) distros available, and even while being not so popular, it still takes [its niche](http://sourcemage.org/About) quite well.

The module provides wide-range functionality for controlling SMGL (so we call them) systems the way you want them to be (pretty minimal and up-to-date), including dependencies management, full/partial system rebuild, updating of the components (package manager, collections of software, packages), so we could provide a universal alternative to our outdated [Cabal command](http://sourcemage.org/Sorcery/Commands/Cabal), used to be a multi-system control tool.

[Intro page](http://sourcemage.org/Intro) is there if you need some more information on how our system works.
You can also find us lurking on #sourcemage at Freenode IRC network and ask a question!

Thanks!

### CODE REVIEW

**Comment by vaygr** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/1040

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (gundalow) [inline lib/ansible/modules/packaging/os/sorcery.py:96]: Can you please rewrite the examples with a 
`name:` and multi line statements using `key: value`
We are in the process of standardising all the modules to use that format.

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (gundalow) [inline lib/ansible/modules/packaging/os/sorcery.py]: To enable the module to work with debuggers can you please rewrite this as:
```
if __name__ == '__main__':
    main()
```

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (gundalow) [inline lib/ansible/modules/packaging/os/sorcery.py]: Could you just import the specific things you need?

**Review by gundalow — CHANGES_REQUESTED** [review]
Just a few minor tweaks needed. Nice to see a module with decent documentation. 

I see @bcoca has a few comments on the previous PR which may be outstanding, so we should get his mark of approval before merging as well.

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (vaygr) [reply lib/ansible/modules/packaging/os/sorcery.py]: absolutely, was thinking about it.

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (vaygr) [reply lib/ansible/modules/packaging/os/sorcery.py:96]: Yeah, I see, done.

File: lib/ansible/modules/packaging/os/sorcery.py
Comment (vaygr) [reply lib/ansible/modules/packaging/os/sorcery.py]: done,

btw, is there any value in making this module more portable so it'd work with older branches of Ansible, e.g. `1.9`?

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Ansible 1.9 will be going out of support very soon. Ansible 2.2 was released recently and I'd strongly suggest upgrading. 

As a general rule we don't backport new modules. However if you include this file in `./library/` directory along side `roles` if will be available to you.

**Comment by gundalow** [comment]
@vaygr if @bcoca has any feedback please include that in a follow up PR.

Thanks for the new module.

**Comment by vaygr** [comment]
will do, thank you.

**Pull request merged**
